tensorflow_recommenders_addons/dynamic_embedding/core/ops/redis_table_ops.cc

```diff
@@ -0,0 +1,252 @@
+    .Attr("Tin: type")
+    .SetShapeFn([](InferenceContext *c) {
+      ShapeHandle handle;
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(0), 0, &handle));
+      TF_RETURN_IF_ERROR(c->WithRankAtLeast(c->input(1), 1, &handle));
+
+      // TODO(turboale): Validate keys shape.
+      return Status::OK();
+    });
+
+REGISTER_OP("TFRA>RedisTableSize")
+    .Input("table_handle: resource")
+    .Output("size: int64")
+    .SetShapeFn(ScalarAndTwoElementVectorInputsAndScalarOutputs);
+
+REGISTER_OP("TFRA>RedisTableExport")
+    .Input("table_handle: resource")
+    .Output("keys: Tkeys")
+    .Output("values: Tvalues")
+    .Attr("Tkeys: type")
+    .Attr("Tvalues: type")
+    .SetShapeFn([](InferenceContext *c) {
+      ShapeHandle handle;
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(0), 0, &handle));
+      ShapeHandle keys = c->UnknownShapeOfRank(1);
+      ShapeAndType value_shape_and_type;
+      TF_RETURN_IF_ERROR(ValidateTableResourceHandle(
+          c,
+          /*keys=*/keys,
+          /*key_dtype_attr=*/"Tkeys",
+          /*value_dtype_attr=*/"Tvalues",
+          /*is_lookup=*/false, &value_shape_and_type));
+      c->set_output(0, keys);
+      c->set_output(1, value_shape_and_type.shape);
+      return Status::OK();
+    });
+
+REGISTER_OP("TFRA>RedisTableImport")
+    .Input("table_handle: resource")
+    .Input("keys: Tin")
+    .Input("values: Tout")
+    .Attr("Tin: type")
+    .Attr("Tout: type")
+    .SetShapeFn([](InferenceContext *c) {
+      ShapeHandle handle;
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(0), 0, &handle));
+
+      ShapeHandle keys;
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(1), 1, &keys));
+      TF_RETURN_IF_ERROR(c->Merge(keys, c->input(2), &keys));
+      return Status::OK();
+    });
+
+Status RedisTableShape(InferenceContext *c, const ShapeHandle &key,
+                       const ShapeHandle &value) {
+  c->set_output(0, c->Scalar());
+
+  ShapeHandle key_s;
+  TF_RETURN_IF_ERROR(c->WithRankAtMost(key, 1, &key_s));
+
+  DataType key_t;
+  TF_RETURN_IF_ERROR(c->GetAttr("key_dtype", &key_t));
+
+  DataType value_t;
+  TF_RETURN_IF_ERROR(c->GetAttr("value_dtype", &value_t));
+
+  c->set_output_handle_shapes_and_types(
+      0, std::vector<ShapeAndType>{{key_s, key_t}, {value, value_t}});
+
+  return Status::OK();
+}
+
+REGISTER_OP("TFRA>RedisTableOfTensors")
+    .Output("table_handle: resource")
+    .Attr("container: string = ''")
+    .Attr("shared_name: string = ''")
+    .Attr("use_node_name_sharing: bool = false")
+    .Attr("key_dtype: type")
+    .Attr("value_dtype: type")
+    .Attr("value_shape: shape = {}")
+    .Attr("embedding_name: string = ''")
+    .Attr("redis_connection_mode: int = 1")
+    .Attr("redis_master_name: string = ''")
+    .Attr("redis_host_ip: string = ''")
+    .Attr("redis_host_port: int = 26379")
+    .Attr("redis_password: string = ''")
+    .Attr("redis_db: int = 0")
+    .Attr("storage_slice: int = 0")
+    .Attr("using_MD5_prefix_name: bool = false")
```

Comment:
maybe change MD5 to 'md5' by following TF style , for example:
https://github.com/tensorflow/tensorflow/blob/27551046f5c49939c508ed7a7c33cb6724aa747c/tensorflow/python/keras/utils/data_utils_test.py#L58